mergify_engine/mergify_pull.py

```diff
@@ -402,6 +402,21 @@ def _merge_failed(self, e):
                        exc_info=True)
         return False
 
+    def delete_head_branch(self):
```

Comment:
I also delete head branch here: https://github.com/Mergifyio/mergify-engine/blob/master/mergify_engine/engine.py#L242

I'm sure we can factorize both.